base/slicearray.jl

```diff
@@ -40,33 +40,36 @@ unitaxis(::AbstractArray) = Base.OneTo(1)
 
 function Slices(A::P, slicemap::SM, ax::AX) where {P,SM,AX}
     N = length(ax)
-    argT = map((a,l) -> l === (:) ? Colon : eltype(a), axes(A), slicemap)
+    parent_axes = ntuple(d -> axes(A, d), length(slicemap))
```

Comment:
I think it'd probably be good to explicitly error if this new invariant (that `slicemap` must be at least `ndims(A)` long) is broken.  I know you're changing the only call-site, but I'd rather get an error than a weird truncation or something.  And I think this should be statically known/removable:

```suggestion
    length(slicemap) >= ndims(A) || throw(AssertionError("the second argument to the Slices constructor must not have fewer elements than there are dimensions"))
    parent_axes = ntuple(d -> axes(A, d), length(slicemap))
```

Reply:
of course I'm down to add such a check, but I don't yet understand how that would be a new problem after this PR (vs existing). on master, if `slicemap` is shorter than `ndims` than the `map` call will truncate to `length(slicemap)` already, right?

Reply:
Ah, you're right. That's great. I always forget that the 2-arg `map` doesn't demand/assert equal lengths.  I was worried this change might introduce a fun new failure mode but you're right this behaves exactly the same way.  So, yeah, take it or leave it :)

Reply:
seems like a decent idea anyway, and master already infers `!n` (which this doesn't change)